Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide.hs

```diff
@@ -86,22 +101,25 @@ cancelRideHandler ServiceHandle {..} requestorId rideId req = withLogTag ("rideI
           driver <- findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
           unless (authPerson.merchantId == driver.merchantId) $ throwError (RideDoesNotExist rideId.getId)
           logTagInfo "admin -> cancelRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId ride.id)
-          buildRideCancelationReason Nothing DBCR.ByMerchant ride
+          buildRideCancelationReason Nothing Nothing Nothing Nothing DBCR.ByMerchant ride
         DP.DRIVER -> do
           unless (authPerson.id == driverId) $ throwError NotAnExecutor
           logTagInfo "driver -> cancelRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId ride.id)
-          buildRideCancelationReason (Just driverId) DBCR.ByDriver ride
+          location <- findDriverLocationId driverId >>= fromMaybeM LocationNotFound
+          booking <- findBookingById ride.bookingId >>= fromMaybeM (BookingNotFound ride.bookingId.getId)
+          locToPickup <- pickUpDistance booking.providerId (getCoordinates location) (getCoordinates booking.fromLocation)
```

Comment:
can you name is `disToPickup` instead of `locToPickup` ?

Reply:
changed.